### PR TITLE
ref: PG affinity_type -> placement_group_type; is_strict -> placement_group_policy

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request implements the following upcoming breaking changes for the VM Placement GA:

* AffinityType -> PlacementGroupType
* IsStrict -> PlacementGroupPolicy

**NOTE: We will need communication around this breaking change in the release notes.**

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally and have pointed your local environment at an API instance where PGs are available:

```
export LINODE_URL=...
export LINODE_TOKEN=...
```

### Integration Testing

```
make ARGS="-run TestPlacementGroup TestInstance_withPG" fixtures
```

### Manual Testing

1. In a linodego sandbox environment (e.g. dx-devenv), run the following:

```go
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/linode/linodego"
)

const targetRegion = "eu-west"

func main() {
	ctx := context.Background()

	client := linodego.NewClient(nil)
	client.SetToken(os.Getenv("LINODE_TOKEN"))

	// Create a PG
	pg, err := client.CreatePlacementGroup(ctx, linodego.PlacementGroupCreateOptions{
		Label:                "test-pg",
		Region:               targetRegion,
		PlacementGroupType:   linodego.PlacementGroupTypeAntiAffinityLocal,
		PlacementGroupPolicy: linodego.PlacementGroupPolicyStrict,
	})
	if err != nil {
		log.Fatal(err)
	}

	// Provision two instances assigned to the PG
	compliantOnly := true

	for i := 0; i < 2; i++ {
		_, err := client.CreateInstance(ctx, linodego.InstanceCreateOptions{
			Region: targetRegion,
			Type:   "g6-nanode-1",
			Label:  fmt.Sprintf("test-instance-%d", i),
			PlacementGroup: &linodego.InstanceCreatePlacementGroupOptions{
				ID:            pg.ID,
				CompliantOnly: &compliantOnly,
			},
		})
		if err != nil {
			log.Fatal(err)
		}
	}
}
```

2. Navigate to Cloud Manager and ensure a new Placement Group has been created with two instances assigned to it.
3. Modify this code with update logic, assignment logic, etc.
4. Re-run the file and ensure everything works as expected.

### CODE REVIEW

File: test/integration/fixtures/TestInstance_withPG.yaml [hunk 1]
```diff
@@ -317,7 +317,7 @@ interactions:
     code: 200
```
Comment (lgarber-akamai) [inline test/integration/fixtures/TestInstance_withPG.yaml:317]: These fixtures will need to be re-ran before merging, I just manually modified the request & response bodies for testing purposes.

**Review by yec-akamai — APPROVED** [review]
Works well on my end!

**Review by ezilber-akamai — APPROVED** [review]
Working as expected!

**Review by ykim-1 — APPROVED** [review]
LGTM, working as expected. Nice work!

**Pull request merged**